pkg/network/node/egressip.go

```diff
@@ -136,11 +149,88 @@ func (eip *egressIPWatcher) ReleaseEgressIP(egressIP, nodeIP string) {
 		if err := eip.releaseEgressIP(egressIP, mark); err != nil {
 			utilruntime.HandleError(fmt.Errorf("Error releasing Egress IP %q: %v", egressIP, err))
 		}
-	} else if eip.vxlanMonitor != nil {
-		eip.vxlanMonitor.RemoveEgressIP(nodeIP, egressIP)
+	} else {
+		eip.removeEgressIP(nodeIP, egressIP)
 	}
 }
 
+func (eip *egressIPWatcher) addEgressIP(nodeIP, egressIP string) {
+	if eip.monitorNodes[nodeIP] != nil {
+		eip.monitorNodes[nodeIP].egressIPs.Insert(egressIP)
+		return
+	}
+	klog.V(4).Infof("Monitoring node %s", nodeIP)
+
+	eip.monitorNodes[nodeIP] = &egressNode{
+		nodeIP:    nodeIP,
+		egressIPs: sets.NewString(egressIP),
+	}
+	if len(eip.monitorNodes) == 1 {
+		eip.stop = make(chan struct{})
+		go utilwait.PollUntil(defaultPollInterval, eip.poll, eip.stop)
+	}
+}
+
+func (eip *egressIPWatcher) removeEgressIP(nodeIP, egressIP string) {
+	if eip.monitorNodes[nodeIP] == nil {
+		return
+	}
+	eip.monitorNodes[nodeIP].egressIPs.Delete(egressIP)
+	if eip.monitorNodes[nodeIP].egressIPs.Len() == 0 {
+		klog.V(4).Infof("Unmonitoring node %s", nodeIP)
+		delete(eip.monitorNodes, nodeIP)
+		if len(eip.monitorNodes) == 0 && eip.stop != nil {
+			close(eip.stop)
+			eip.stop = nil
+		}
+	}
+}
+
+func (eip *egressIPWatcher) poll() (bool, error) {
```

Comment:
in `egressVXLANMonitor` we hold the evm mutex around all of `poll`, which is bad because of the sleep-and-retry... it would be better to only hold it around `check`. (Although that's also not ideal now, since doing all the pings could take a while... but I guess `addNodeIP` and `removeNodeIP` don't get called that often, so there won't be that many situations where they have to wait for a `check` to finish and release the mutex, so we can survive with this for now.)